src/main/scala/beam/sim/BeamMobsim.scala

```diff
@@ -194,7 +194,7 @@ class BeamMobsim @Inject()(
       val modesAvailable: List[BeamMode] = nonCavModesAvailable ++ cavModeAvailable
 
       persons.foreach { person =>
-        if (matsimServices.getIterationNumber == 0) {
+        if (matsimServices.getIterationNumber.intValue() == 0) {
```

Comment:
If we don't use this .intValue() then a warning is produced? This decreases readability a little.

Reply:
@dimaopen scapegoat gets lost because `getIterationNumber` returns `java.lang.Integer` and `intValue` returns `int` which it considered the same as scala `Int`. So, this is needed in order to avoid warning

Reply:
Options I can see:
```Scala
matsimServices.getIterationNumber.intValue() == 0
```
or
```Scala
matsimServices.getIterationNumber == java.lang.Integer.valueOf(0)
```
or
```Scala
val ZERO = java.lang.Integer.valueOf(0)
...
matsimServices.getIterationNumber == ZERO
```
or
```Scala
Objects.equals(matsimServices.getIterationNumber, 0)
```
or
```Scala
0.equals(matsimServices.getIterationNumber)
```

Reply:
I think the last one is best here 0.equals...

Reply:
But it is more of a preference - so nbd either way